apps/emqx_bridge_influxdb/src/emqx_bridge_influxdb_connector.erl

```diff
@@ -711,56 +716,111 @@ time_unit(ns) -> nanosecond.
 maps_config_to_data(K, V, {Data, Res}) ->
     KTransOptions = #{return => rawlist, var_trans => fun key_filter/1},
     VTransOptions = #{return => rawlist, var_trans => fun data_filter/1},
-    NK0 = emqx_placeholder:proc_tmpl(K, Data, KTransOptions),
+    NK = emqx_placeholder:proc_tmpl(K, Data, KTransOptions),
     NV = emqx_placeholder:proc_tmpl(V, Data, VTransOptions),
-    case {NK0, NV} of
+    case {NK, NV} of
         {[undefined], _} ->
             {Data, Res};
         %% undefined value in normal format [undefined] or int/uint format [undefined, <<"i">>]
         {_, [undefined | _]} ->
             {Data, Res};
         _ ->
-            NK = list_to_binary(NK0),
-            {Data, Res#{NK => value_type(NV)}}
+            {Data, Res#{
+                list_to_binary(NK) => value_type(NV, tmpl_type(V))
+            }}
     end.
 
-value_type([Int, <<"i">>]) when
-    is_integer(Int)
-->
+value_type([Int, <<"i">>], mixed) when is_integer(Int) ->
     {int, Int};
-value_type([UInt, <<"u">>]) when
-    is_integer(UInt)
-->
+value_type([UInt, <<"u">>], mixed) when is_integer(UInt) ->
     {uint, UInt};
 %% write `1`, `1.0`, `-1.0` all as float
 %% see also: https://docs.influxdata.com/influxdb/v2.7/reference/syntax/line-protocol/#float
-value_type([Number]) when is_number(Number) ->
-    Number;
-value_type([<<"t">>]) ->
+value_type([Number], _) when is_number(Number) ->
+    {float, Number};
+value_type([Str], variable) when is_binary(Str) ->
+    Str;
+value_type([Str], literal) when is_binary(Str) ->
+    %% if Str is a literal string suffixed with `i` or `u`, we should convert it to int/uint.
+    %% otherwise, we should convert it to float.
+    NumStr = binary:part(Str, 0, byte_size(Str) - 1),
+    case binary:part(Str, byte_size(Str), -1) of
+        <<"i">> ->
+            maybe_convert_to_integer(NumStr, Str, int);
+        <<"u">> ->
+            maybe_convert_to_integer(NumStr, Str, uint);
+        _ ->
+            maybe_convert_to_float_str(Str)
```

Comment:
maybe allow quotes around the string so it's always string even if it's a float ?

Reply:
Fixed